py_neuromodulation/nm_mnelsl_generator.py

```diff
@@ -0,0 +1,87 @@
+from mne_lsl.player import PlayerLSL
+import numpy as np
+import mne
+from pathlib import Path
+
+from py_neuromodulation import logger, nm_types
+
+
+class LSLOfflinePlayer:
+
+    def __init__(
+        self,
+        stream_name: str = "lsl_offline_player",
+        f_name: str | nm_types.PathLike = None,
+        sfreq: int | float | None = None,
+        data: np.ndarray | None = None,
+    ) -> None:
+        """Initialization of MNE-LSL offline player.
+        Either a filename (PathLike) is provided,
+        or data and sampling frequency to initialize an example mock-up stream.
+
+
+        Parameters
+        ----------
+        settings : dict
+            _description_
+        stream_name : str, optional
+            _description_, by default "example_stream"
+        f_name : str | None, optional
+            _description_, by default None
+        sfreq : int | float | None, optional
+            _description_, by default None
+        data : np.ndarray | None, optional
+            _description_, by default None
+
+        Raises
+        ------
+        ValueError
+            _description_
+        """
+        self.sfreq = sfreq
+        self.stream_name = stream_name
+        got_fname = f_name is not None
+        got_sfreq_data = sfreq is not None and data is not None
+        if not (got_fname or got_sfreq_data):
+            error_msg = "Either f_name or sfreq and data must be provided."
+            logger.critical(error_msg)
+            raise ValueError(error_msg)
+
+        if f_name is not None:
+            self._path_raw = f_name
+
+        if sfreq is not None and data is not None:
+
+            info = mne.create_info(
+                ch_names=[f"ch{i}" for i in range(data.shape[0])],
+                ch_types=["dbs" for _ in range(data.shape[0])], # TODO Maybe change this to auto detect channel type? Or make it a parameter?
```

Comment:
In case `f_name is not None` we auto detect ch_types. But when working with `sfreq` and `data` we don't give an option to set the ch_types. so I would set ch_types as an optional parameter when initialising the player. 
Should this overwrite the ch_types in case` f_name is not None` ? 
What should the default be? "dbs" or `None`?

Reply:
I think the `ch_types` should not overwrite the ones stored in `f_name`. And having `None` would be a good default. I think if you don't provide the ch_type, the LSL stream default was "eeg" right?